### PR TITLE
fix(model): check attribute overrides when defining assoc mixins (#9952)

### LINKED ISSUE #9952: Association mixins attempt to override attributes on model, causing error
## What are you doing?

When I have a model that defines a property whose name matches one of the mixin names that would get generated when I add an association to the model, the model definition and association blows up.

```js
module.exports = (sequelize, DataTypes) => {
  const Company = sequelize.define("company", {
    hasLocations: {
      type: DataTypes.BOOLEAN
    }
  })
  Company.associate = models => {
    Company.Locations = Company.hasMany(models.location)
  }
  return Company
}

```

## What do you expect to happen?
I don't want it to blow up and instead want to skip the `hasLocations()` method mixin definition because it conflicts with the `hasLocations` prop

## What is actually happening?
I get the following error

```sh
TypeError: Cannot read property 'hasLocations' of undefined
    at Model.get (<project-root>/node_modules/sequelize/lib/model.js:3305:29)
    at Model.funcs.(anonymous function) (<project-root>/node_modules/sequelize/lib/model.js:1022:25)
    at Object.mixinMethods (<project-root>/node_modules/sequelize/lib/associations/helpers.js:58:13)
    at HasMany.mixin (<project-root>/node_modules/sequelize/lib/associations/has-many.js:155:13)
    at Function.hasMany (<project-root>/node_modules/sequelize/lib/associations/mixin.js:34:17)
```

__Sequelize version:__ 5.0.0-beta.12
__Tested with latest release:__ Yes (If yes, specify that version)

### PR DESCRIPTION
**Pull request by bkoltai**
### Description of change

Closes #9952

### CODE REVIEW

File: lib/associations/helpers.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ function mixinMethods(association, obj, methods, aliases) {
 
   for (const method of methods) {
     // don't override custom methods
-    if (!obj[association.accessors[method]]) {
+    if (!obj.hasOwnProperty(association.accessors[method]) && !obj[association.accessors[method]]) {
```
Comment (sushantdhiman) [inline lib/associations/helpers.js]: Just `!obj.hasOwnProperty(association.accessors[method])` should be enough

**Comment by bkoltai** [comment]
I thought so too but when I had that a bunch of other tests failed. I
believe it has something to do with the dynamic nature of some of the
method calls on models

**Comment by sushantdhiman** [comment]
Just change this, I'll see if i can fix those tests

**Comment by bkoltai** [comment]
Ok, I've updated that line to just use the `hasOwnProperty`

**Comment by sushantdhiman** [comment]
Looks like all tests are passing, merging

**Pull request merged**